crates/router/src/core/payments/helpers.rs

```diff
@@ -2059,7 +2061,7 @@ pub async fn make_pm_data<'a, F: Clone, R>(
 
         (Some(_), _) => {
             let (payment_method_data, payment_token) = payment_methods::retrieve_payment_method(
-                request,
+                &request.map(Into::into),
```

Comment:
You can just pass `request` if u make this a reference above. No type conversion required as request type is already `Option<domain::PaymentMethodData>`